File: Backend/src/main/java/com/inn/cstore/service/ClienteService.java
package com.inn.cstore.service;

import java.util.List;
import java.util.Map;

import org.springframework.http.ResponseEntity;

import com.inn.cstore.wrapper.ClienteWrapper;

public interface ClienteService {

    // Lista todos los clientes con su resumen de compras
    ResponseEntity<List<ClienteWrapper>> getAllClientes();

    // Detalle completo de un cliente (incluye historial de facturas)
    ResponseEntity<ClienteWrapper> getClienteResumen(Integer id);

    // Agregar cliente manualmente
    ResponseEntity<String> addCliente(Map<String, Object> requestMap);

    // Actualizar datos del cliente
    ResponseEntity<String> updateCliente(Map<String, Object> requestMap);

    // Eliminar cliente
    ResponseEntity<String> deleteCliente(Integer id);

}

File: Frontend/cstore-fronted/src/app/core/services/cliente.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../enviroments/enviroment';
import { TokenService } from './token.service';

@Injectable({
  providedIn: 'root',
})
export class ClienteService {
  private apiUrl = `${environment.apiUrl}/cliente`;

  constructor(private http: HttpClient, private tokenService: TokenService) {}

  private getHeaders(): HttpHeaders {
    const token = this.tokenService.getToken();
    return new HttpHeaders({
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`,
    });
  }

  // Lista todos los clientes con resumen de compras
  getAllClientes(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/getAllClientes`, {
      headers: this.getHeaders(),
    });
  }

  // Detalle completo de un cliente (historial de facturas)
  getClienteResumen(id: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/getResumen/${id}`, {
      headers: this.getHeaders(),
    });
  }
}
